tests/FSharp.Compiler.ComponentTests/EmittedIL/FixedBindings/PinTypeWithExtensionGetPinnableReferenceReturningByref.fs

```diff
@@ -0,0 +1,22 @@
+module FixedBindings
+open Microsoft.FSharp.NativeInterop
+open System.Runtime.CompilerServices
+
+type RefField<'T> = { mutable _value: 'T }
+
+[<Extension>]
+type RefFieldExtensions =
+    [<Extension; MethodImpl(MethodImplOptions.NoInlining)>]
+    static member GetPinnableReference(refField: RefField<'T>) : byref<'T> = &refField._value 
```

Comment:
Do we expect F# type augmentaitons work too?

For example, 
```fsharp
type RefField<'T> = { mutable _value: 'T }
type RefField<'T> with
    member _.GetPinnableReference(refField: RefField<'T>) : byref<'T> = &refField._value
```

Reply:
Yep, added a test: https://github.com/dotnet/fsharp/blob/557ef00ff1119c1c7c97a7ef92cf3d913c85cfa8/tests/FSharp.Compiler.ComponentTests/EmittedIL/FixedBindings/PinTypeWithFSharpStyleExtensionGetPinnableReferenceReturningByref.fs